mathesar_ui/src/component-library/pagination/Pagination.svelte

```diff
@@ -74,28 +78,28 @@
             1
           </a>
         {:else}
-          <span
+          <button
             tabindex="0"
             role="link"
             aria-label="Goto Page 1"
             class="page"
             on:click={(e) => setPage(e, 1)}
           >
             1
-          </span>
+          </button>
         {/if}
       </li>
       {#if pageInfo.start > 2}
         <li>
-          <span
+          <button
             tabindex="0"
             role="link"
             aria-label="Goto Page {pageInfo.prevPageWindow}"
             on:click={(e) => setPage(e, pageInfo.prevPageWindow)}
           >
-            <Icon class="ellipsis" data={faEllipsisH} />
+            <Icon class="ellipsis" tabindex="-1" data={faEllipsisH} />
```

Comment:
Is there a reason why `tabindex` is set for icons?

Reply:
@pavish Yes, when an ellipsis is shown and one uses the tab key to navigate the pagination component, this prevents the focus from disappearing for a single click. I suspect it disappears because it focuses the arrow, which is not shown on the page.  
The only thing that confuses me a little is that the only way it works is if `tabindex="-1"` is set on the ellipsis and not on the arrow which should be the one that is not focusable since it is not showing on the page. I tried doing it with CSS but didn't work.

Reply:
Let me try to understand this better. I think the following is the case:
1. Ellipsis is shown and user clicks on it (i.e the arrow).
2. The page number changes but ellipsis/arrow is now not present because it is not required. Eg., Ellipsis is present when active page is 7. Clicking on it changes active page to 2, and ellipsis is no longer present.
3. Since the last focused element is no longer present on the dom, the focus state is reset by the browser.

If you're referring to this problem, we should be resolving it by manually setting focus on the current active page when ellipsis/arrow disappears. In the above example, when we click on the ellipsis and page number is changed to 2, we should be doing `element.focus()` to focus the button with page number 2. You can add a `data-page` attribute to help query the element.

Reply:
@pavish I'm not referring to this problem. I'll make a loom video showing you what it does, it will be easier. 
For the problem that you have mentioned, isn't that the standard behavior for [Ant Design](https://ant.design/components/pagination/)?

Reply:
Thanks, I'll take a look at the loom video once you post it.

Regarding the problem I've described, Ant Design may not have handled it, but it'll be better for us to handle it. We don't have to do that in this PR. Please feel free to open an issue linking to the above comment and we can address it later.

Reply:
@pavish [Video Explaining the problem I described](https://www.loom.com/share/49c31d28668543f9865a4da6c33d1c03)

Reply:
@pavish Sounds good!

Reply:
@mr-gabe49 Interesting! I didn't notice this behavior on Firefox earlier so I wasn't sure what the issue was. Seems like this issue is present in both Chrome and Safari.

This is because both Blink and Webkit find any svg with a focus event handler focusable. So this issue is present wherever we use icons on our application. Nice catch.

I don't think our `Icon` component needs to handle focus events or any event for that matter. Please remove all `on:<event>` handlers within our `Icon` component. That should fix this issue.

Reply:
@pavish Sounds good, I'll make the changes.

Reply:
Created issue https://github.com/centerofci/mathesar/issues/1040#issue-1122334518